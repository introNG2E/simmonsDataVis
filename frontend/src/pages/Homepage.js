import React, { useState } from 'react';
import {Redirect} from 'react-router-dom';
import {Grid, Image, Header, Container, Button, Divider, Segment, GridColumn} from 'semantic-ui-react'
import '../styles/Homepage.css'

function Homepage(props) {

    const [toData, setToData] = useState(false);
    const [toSurvey, setToSurvey] = useState(false);

    function onDataPress() {
        props.history.push('/data');
    }

    function onSurveyPress() {
        props.history.push('/survey')
    }

    console.log(props.user)

    return (
        <div className="homePage">
            <Divider/>
            <Header as='h1' textAlign='center'>Engineering Out-Of-Class Activity Decision Support System</Header>
            <Divider/>
            <Container text>
                <Segment inverted color="grey">
                    <p>The OOCA Decision Support System aims at guiding students (i.e. especially
                        undergraduate engineering) on making decisions on professional development plan
                        and support educators, program managers and policy makers on developing
                        strategic plan for engineering programs. This system is developed as part of the
                        NSF project: CAREER:Investigating Co-Curricular Participation of Students
                        Underrepresented in Engineering. The full project description can be found on: &nbsp;
                        <a href="https://www.nsf.gov/awardsearch/showAward?AWD_ID=1351156&HistoricalAwards=false">here
                        </a>.</p>
                    <p>This system uses public data and survives by your contributions. Please take your time
                        to complete data the survey. By doing so, you are helping to improve the system. Please note
                        that
                        you need to be at least 18 years old to take this survey.</p>
                </Segment>
            </Container>
            <Grid columns='2'>
                <Grid.Column>
                    <Button 
                        onClick={onDataPress} 
                        size='massive'
                        className='router'>
                        &nbsp; Data &nbsp;
                    </Button>
                </Grid.Column>
                <Grid.Column>
                    <Button 
                        onClick={onSurveyPress} 
                        size='massive'
                        className='router'>
                        Survey
                    </Button>
                </Grid.Column>
            </Grid>
            <Divider/>
            <Divider/>
            <Divider/>
        </div>
    );
}

export default Homepage;
