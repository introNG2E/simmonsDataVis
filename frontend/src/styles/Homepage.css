.homePage {
    

    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    max-width: 85vw;
}

a {
    color:#00ccff;
}