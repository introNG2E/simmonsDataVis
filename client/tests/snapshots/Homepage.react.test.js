import React from 'react';
import Homepage from '../../src/pages/Homepage';
import renderer from 'react-test-renderer';

it('renders correctly with id passed to it', async () => {
  const tree = renderer
    .create(<Homepage getId={'5de56e20ed26b55db0ba44c4'} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly without id passed to it', async () => {
  const tree = renderer
    .create(<Homepage getId={''} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});