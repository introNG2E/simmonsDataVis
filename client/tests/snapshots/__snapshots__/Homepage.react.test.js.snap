// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with id passed to it 1`] = `
<div
  className="homePage"
>
  <div
    className="ui divider"
  />
  <h1
    className="ui center aligned header"
  >
    Engineering Out-Of-Class Activity Decision Support System
  </h1>
  <div
    className="ui divider"
  />
  <div
    className="ui grey inverted very padded segment"
  >
    <p>
      The OOCA Decision Support System aims at guiding students (i.e. especially undergraduate engineering) on making decisions on professional development plan and support educators, program managers and policy makers on developing strategic plan for engineering programs. This system is developed as part of the NSF project: CAREER:Investigating Co-Curricular Participation of Students Underrepresented in Engineering. The full project description can be found on:  
      <a
        href="https://www.nsf.gov/awardsearch/showAward?AWD_ID=1351156&HistoricalAwards=false"
      >
        here
      </a>
      .
    </p>
    <p>
      This system uses public data and survives by your contributions. Please take your time to complete data the survey. By doing so, you are helping to improve the system. Please note that you need to be at least 18 years old to take this survey.
    </p>
  </div>
  <div
    className="ui divider"
  />
  <div
    className="ui divider"
  />
  <div
    className="ui two column grid"
  >
    <div
      className="column"
    >
      <button
        className="ui massive button router"
        onClick={[Function]}
      >
          Data  
      </button>
    </div>
    <div
      className="column"
    >
      <button
        className="ui massive button router"
        onClick={[Function]}
      >
        Survey
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly without id passed to it 1`] = `
<div
  className="homePage"
>
  <div
    className="ui divider"
  />
  <h1
    className="ui center aligned header"
  >
    Engineering Out-Of-Class Activity Decision Support System
  </h1>
  <div
    className="ui divider"
  />
  <div
    className="ui grey inverted very padded segment"
  >
    <p>
      The OOCA Decision Support System aims at guiding students (i.e. especially undergraduate engineering) on making decisions on professional development plan and support educators, program managers and policy makers on developing strategic plan for engineering programs. This system is developed as part of the NSF project: CAREER:Investigating Co-Curricular Participation of Students Underrepresented in Engineering. The full project description can be found on:  
      <a
        href="https://www.nsf.gov/awardsearch/showAward?AWD_ID=1351156&HistoricalAwards=false"
      >
        here
      </a>
      .
    </p>
    <p>
      This system uses public data and survives by your contributions. Please take your time to complete data the survey. By doing so, you are helping to improve the system. Please note that you need to be at least 18 years old to take this survey.
    </p>
  </div>
  <div
    className="ui divider"
  />
  <div
    className="ui divider"
  />
  <div
    className="ui two column grid"
  >
    <div
      className="column"
    >
      <button
        className="ui massive button router"
        onClick={[Function]}
      >
          Data  
      </button>
    </div>
    <div
      className="column"
    >
      <button
        className="ui massive button router"
        onClick={[Function]}
      >
        Survey
      </button>
    </div>
  </div>
</div>
`;
